src/shims/unix/macos/foreign_items.rs

```diff
@@ -67,6 +67,12 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
                 let result = this.ftruncate64(fd, length)?;
                 this.write_scalar(Scalar::from_i32(result), dest)?;
             }
+            "realpath" | "realpath$DARWIN_EXTSN" => {
```

Comment:
```suggestion
            "realpath$DARWIN_EXTSN" => {
```
That variant is dead code, since it is already handled in the general unix method.